app.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'csv'
+
+set :strict_paths, false
+
+before do
+  data_dir = File.join(Dir.pwd, 'data')
+  @csv_path = File.join(data_dir, 'memos.csv')
+  @max_id_path = File.join(data_dir, 'max_id.txt')
+
+  Dir.mkdir(data_dir) unless Dir.exist?(data_dir)
+  prepare_max_id_file(@max_id_path)
+  prepare_data_file(@csv_path)
+end
+
+get '/memos' do
+  @page_title = 'メモ一覧'
+  @memos = []
+
+  CSV.foreach(@csv_path, headers: true) do |memo|
+    @memos << { id: memo['id'], title: memo['title'] }
+  end
+  erb :memos
+end
```

Comment:
意味的に以下のようなまとまりにするのがよいと思います。コメントは補足のためのものなので残さなくてよいです。

```suggestion
get '/memos' do
  # メモの取得
  @memos = []
  CSV.foreach(@csv_path, headers: true) do |memo|
    @memos << { id: memo['id'], title: memo['title'] }
  end

  # ページの表示
  @page_title = 'メモ一覧'
  erb :memos
end
```

Reply:
同様の箇所も含めて、そのように修正します。